crates/sui/src/upgrade_compatibility/mod.rs

```diff
@@ -58,46 +66,46 @@ pub(crate) enum UpgradeCompatibilityModeError {
     },
     StructAbilityMismatch {
         name: Identifier,
-        old_struct: Struct,
-        new_struct: Struct,
+        old_struct: Rc<Struct>,
```

Comment:
Small perf/size improvements :) Before this, the entire thing was being cloned into this error